src/spyglass/utils/dj_mixin.py

```diff
@@ -697,25 +706,102 @@ def super_delete(self, warn=True, *args, **kwargs):
-        spawn processes in their make function
+        spawn processes in their make function and always use transactions.
+
+        `_use_transaction` class attribute can be set to False to disable
+        transaction protection for a table. This is not recommended for tables
+        with short processing times. A before-and-after hash check is performed
+        to ensure upstream tables have not changed during populate, and may
+        be a more time-consuming process. To permit the `make` to insert without
+        populate, set `_allow_insert` to True.
         """
-
-        # Pass through to super if not parallel in the make function or only a single process
         processes = kwargs.pop("processes", 1)
+
+        # Decide if using transaction protection
+        use_transact = kwargs.pop("use_transation", None)
+        if use_transact is None:  # if user does not specify, use class default
+            use_transact = self._use_transaction
+            if self._use_transaction is False:  # If class default is off, warn
+                logger.warning(
+                    "Turning off transaction protection this table by default. "
+                    + "Use use_transation=True to re-enable.\n"
+                    + "Read more about transactions:\n"
+                    + "https://docs.datajoint.io/python/definition/05-Transactions.html\n"
+                    + "https://github.com/LorenFrankLab/spyglass/issues/1030"
+                )
+        if use_transact is False and processes > 1:
+            raise RuntimeError(
+                "Must use transaction protection with parallel processing.\n"
+                + "Call with use_transation=True.\n"
+                + f"Table default transaction use: {self._use_transaction}"
+            )
+
+        keys = [True]  # Get keys to pop, needed for no-transact or parallel
+        if use_transact is False or processes > 1 or self._parallel_make:
+            keys = (self._jobs_to_do(restrictions) - self.target).fetch(
+                "KEY", limit=kwargs.get("limit", None)
+            )
+
+        if use_transact is False:
+            upstream_hash = self._hash_upstream(keys)
+            if kwargs:  # Warn of ignoring populate kwargs, bc using `make`
+                logger.warning(
+                    "Ignoring kwargs when not using transaction protection."
+                )
+
         if processes == 1 or not self._parallel_make:
-            kwargs["processes"] = processes
-            return super().populate(*restrictions, **kwargs)
+            if use_transact:  # Pass single-process populate to super
+                kwargs["processes"] = processes
+                return super().populate(*restrictions, **kwargs)
+            else:  # No transaction protection, use bare make
+                for key in keys:
+                    self.make(key)
+                if upstream_hash != self._hash_upstream(keys):
```

Comment:
I think it could be made simpler for the user to clean up the mismatched hash insert by either:
- printing the key in the raised error
- automatically deleting the key before the error (enforces integrity)
- pass the hash result to the make function and do the check in there before `insert` if there's a hash (enforces integrity, requires more edits to the code)

Reply:
Only using the hash makes it tough to determine where the mismatch occurred. I made a commit to delete all the keys and ask the user to start over. It's not ideal, but, from the code I've seen, folks primarily run one key at a time. It seems like an unlikely case that there would be a mismatch - and that it'll have a serious impact in the timeline between now and a new DJ feature we can use

Would you rather I run a table-wise comparison across the two `RestrGraph` objects to nail down where the mismatch occurred?

Reply:
I think the overly cautious solution of delete them all is fine for now.  As you said, it's unlikely that it would occur, and ensuring consistency is a valid priority.  User's shouldn't notice it, and if they do we would need to be going back into the code to figure out why anyways